selfdrive/car/ford/values.py

```diff
@@ -111,14 +111,24 @@ def init_make(self, CP: car.CarParams):
   requests=[
     # CAN and CAN FD queries are combined.
     # FIXME: For CAN FD, ECUs respond with frames larger than 8 bytes on the powertrain bus
+    # TODO: properly handle auxiliary requests to separate queries and add whitelists
     Request(
       [StdQueries.TESTER_PRESENT_REQUEST, StdQueries.MANUFACTURER_SOFTWARE_VERSION_REQUEST],
       [StdQueries.TESTER_PRESENT_RESPONSE, StdQueries.MANUFACTURER_SOFTWARE_VERSION_RESPONSE],
+      # whitelist_ecus=[Ecu.abs, Ecu.engine, Ecu.eps, Ecu.fwdCamera, Ecu.fwdRadar, Ecu.shiftByWire],
+      bus=0,
+      auxiliary=True,
+    ),
+    Request(
+      [StdQueries.MANUFACTURER_SOFTWARE_VERSION_REQUEST],
+      [StdQueries.MANUFACTURER_SOFTWARE_VERSION_RESPONSE],
+      # whitelist_ecus=[Ecu.abs, Ecu.engine, Ecu.eps],
       bus=0,
       auxiliary=True,
```

Comment:
Have you tested to see what this is caused by? Is our query happening too early in the ECU's startup? Or does it not respond to tester present consistently for some reason?

Reply:
I don't think it's related to timing with the ECU startup since I reproduced it with the comma power unplugged - so the 3X started up after the car was already running..
Without this fix, I see the Mazda query returns the same ECUs that are missing from our request, so I imagine some query that happens between Ford and Mazda must put the ECUs in a state where they will respond?

Reply:
I assume the order of queries is consistent (I haven't looked at that closely - do you think it sounds possible?)

Reply:
It seems consistent that you have to send the request twice for some reason - without adding this second request to Ford the ECUs respond to the (exact same) request for Mazda instead

Reply:
Mazda isn't the exact same, there's no tester present. We also queried on the OBD port first, so there could be something related to querying on the OBD port. We know some Hyundai OBD gateways handle the ISO TP communication for you, maybe it needs to do something special to ensure reliable communication, have you checked the bus 0 CAN data when you query on bus 1?

Also, it actually doesn't respond with any sort of ISO TP frames when this happens, error or successful which is odd. Some ECU bug? I didn't see any dropped messages in the Bronco's case.

Reply:
Interesting, it can happen both ways. Here we send tester present on bus 0/1 which EPS responds to, then a little while later try again on the OBD port for the FW query which it ignores, then again later on bus 0 it responds:

![image](https://github.com/commaai/openpilot/assets/25857203/951f6597-c0f9-4358-8a52-6d41d0eee060)

Wonder if it's a time based condition and just sending two tester presents very close together makes it reliable.